symex-evil.el

```diff
@@ -49,6 +49,50 @@
   :enable (normal)
   :exit-hook (symex-exit-mode))
 
+(defun symex-evil-repeat-pre-hook-advice (&rest _)
+  "Prepare the current command for recording the repetition.
+
+This function is meant to advise `evil-repeat-pre-hook' which
+starts recording a repeation during the `pre-command-hook', but
+only records a repition when in normal or visual state. This
+calls `evil-repeat-start' if the buffer is currently in symex
+state."
+  (when evil-local-mode
+    (let ((repeat-type (evil-repeat-type this-command nil)))
+      (when (and (not (evil-repeat-force-abort-p repeat-type))
+                 (not (or (null repeat-type)
+                          (evil-mouse-events-p (this-command-keys))))
```

Comment:
Using boolean algebra, I believe this logic can be simplified a bit, to:
```
(when (and repeat-type
           (not (evil-repeat-force-abort-p repeat-type))
           (not (evil-mouse-events-p (this-command-keys)))
           (evil-symex-state-p))
  (evil-repeat-start))
```

Same for the post-hook advice as well.

Also, nit: it might be worth adding a comment explaining what is happening here, if you think you can summarize it. Not essential though.